File: LeetCode/Easy/0028_Implement_Strstr/Solution.java
// 2021-08-27
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0) return 0;
        if (needle.length() > haystack.length()) return -1;
        
        for (int i = 0; i < haystack.length() - (needle.length() - 1); i++) {
            String word = haystack.substring(i, i + needle.length());
            if (word.equals(needle)) return i;
        }
        return -1;    
    }
}

File: LeetCode/Easy/0028_Implement_Strstr/Solution.cpp
// 2023-03-11
class Solution {
public:
    int strStr(string haystack, string needle) {
        /*
        input: string haystack and string needle to find in haystack
        output: index of needle in haystack

        notes:
        - simply iterate through and substring portions of haystack to 
          compare with needle
        
        analysis:
        - time = O(m*n) where m=length of haystack and n=length of needle
        - space = O(1)
        */
        int m = haystack.length(), n = needle.length();
        for (int i=0; i+n <= m; i++) {
            if (haystack.substr(i, n) == needle)
                return i;
        }
        return -1;
    }
};